### PR TITLE
[Merged by Bors] - Support more env ref types

### PR DESCRIPTION
**Pull request by nacardin**
* Add debugging
* Improve errors by including response from API server

### CODE REVIEW

File: src/k8-metadata-client/Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 edition = "2018"
 name = "k8-metadata-client"
-version = "4.0.1"
+version = "4.0.2"
```
Comment (sehz) [inline src/k8-metadata-client/Cargo.toml:4]: let's bump to `5.0.0` just in case.

**Review by sehz — CHANGES_REQUESTED** [review]
LGTM.  One minor change is probably necessary

File: src/k8-metadata-client/Cargo.toml (hunk 1 above)
Comment (nacardin) [reply src/k8-metadata-client/Cargo.toml]: done

**Review by sehz — APPROVED** [review]
Ship it!

**Pull request closed without merging**